hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl

```diff
@@ -177,15 +185,60 @@ switch ${fpga_family} {
 # A hack that works is to pretend the data width is actually 40 bits. Updatemem
 # seems to write that extra zero bit into the ether without complaint.
 set rom_brams [split [get_cells -hierarchical -filter " PRIMITIVE_TYPE =~ ${bram_regex} && NAME =~ *u_rom_ctrl*"] " "]
-generate_mmi "rom.mmi" $rom_brams "RAMB36" 40 1 1
+dict set memInfo rom brams $rom_brams
+dict set memInfo rom mem_type_regex $mem_type_regex
+dict set memInfo rom fake_word_width 40
+dict set memInfo rom addr_end_multiplier 1
 
 # OTP does not require faking the word width, but it has its own quirk. It seems
 # each 22-bit OTP word is followed by 15 zero words. The MMI's <AddressSpace>
 # and <AddressRange> tags need to account for this or else updatemem will think
 # that its data input overruns the address space. The workaround is to pretend
 # the address space is 16 times larger than we would normally compute.
 set otp_brams [split [get_cells -hierarchical -filter " PRIMITIVE_TYPE =~ ${bram_regex} && NAME =~ *u_otp_ctrl*"] " "]
-generate_mmi "otp.mmi" $otp_brams "RAMB18" 0 16 2
+dict set memInfo otp brams $otp_brams
+dict set memInfo otp mem_type_regex $mem_type_regex
+dict set memInfo otp fake_word_width 0
+dict set memInfo otp addr_end_multiplier 16
+
+# The flash banks have 76-bit wide words. 64 bits are data, and 12 bits are metadata.
```

Comment:
Will `updatemem` support 76-bit wide words per line? It seems like it should, but I am curious to know if you have spotted any quirks when looking at the synthesis results.

Reply:
I think it should, but I don't currently have an image that lets me check. Perhaps next we should cobble together something to test! :)